clients/drcachesim/tracer/tracer.cpp

```diff
@@ -2395,9 +2414,56 @@ event_pre_syscall(void *drcontext, int sysnum)
 #endif
     if (file_ops_func.handoff_buf == NULL)
         memtrace(drcontext, false);
+#ifdef BUILD_PT_TRACER
+    if (op_offline.get_value() && op_enable_kernel_tracing.get_value()) {
+        if (data->syscall_pt_trace.get_recording_sysnum() != -1) {
+            ASSERT(false, "last tracing isn't stopped");
+            return false;
+        }
+        if (!data->syscall_pt_trace.start_syscall_pt_trace(sysnum)) {
+            ASSERT(false, "failed to start syscall pt trace");
+            return false;
+        }
+    }
+#endif
     return true;
 }
 
+static void
+event_post_syscall(void *drcontext, int sysnum)
+{
+#ifdef BUILD_PT_TRACER
+    if (!op_offline.get_value() || !op_enable_kernel_tracing.get_value()) {
+        return;
+    }
+    per_thread_t *data = (per_thread_t *)drmgr_get_tls_field(drcontext, tls_idx);
+
+    if (data->syscall_pt_trace.get_recording_sysnum() == -1) {
+        ASSERT(false, "last syscall is not traced");
+        return;
+    }
+
+    ASSERT(data->syscall_pt_trace.get_recording_sysnum() == sysnum,
+           "last tracing isn't stopped");
+    if (!data->syscall_pt_trace.stop_syscall_pt_trace()) {
+        ASSERT(false, "failed to stop syscall pt trace");
+        return;
+    }
+
+    /* Write a marker to userspace raw trace. */
+    if (tracing_disabled.load(std::memory_order_acquire) != BBDUP_MODE_TRACE)
```

Comment:
So we gather the syscall trace even if we're not tracing userspace code? Add a comment.

Reply:
I misread the code. sorry.

So we do look at `tracing_disabled` in the pre-syscall event. In this post-syscall event though, we're calling `stop_syscall_pt_trace` before we check it.

Reply:
Done.